File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/world/ecs/components/ConversationComponent.java
package de.limbusdev.guardianmonsters.fwmengine.world.ecs.components;

import com.badlogic.ashley.core.Component;

/**
 * Created by georg on 02.12.15.
 */
public class ConversationComponent implements Component {
    /* ............................................................................ ATTRIBUTES .. */
    public String text;
    public String name;
    /* ........................................................................... CONSTRUCTOR .. */
    public ConversationComponent(String text) {
        this(text, "");
    }

    public ConversationComponent(String text, String name) {
        this.text = text;
        this.name = name;
    }
    
    /* ............................................................................... METHODS .. */
    
    /* ..................................................................... GETTERS & SETTERS .. */
}

File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/world/ecs/components/ConversationComponent.kt
package de.limbusdev.guardianmonsters.fwmengine.world.ecs.components

import de.limbusdev.guardianmonsters.fwmengine.world.ecs.LimbusBehaviour

/**
 * ConversationComponent
 *
 * @author Georg Eckert 2015-12-02
 */
class ConversationComponent(var name : String = "", var text : String = "") : LimbusBehaviour()
{
    companion object
    {
        const val className ="ConversationComponent"
        val defaultJson =
                """
                    enabled: true,
                    name: "",
                    text: ""
            """.trimMargin()
    }

    /** Callbacks. Run when the whole conversation is complete. */
    val onConversationFinished = mutableSetOf<(() -> Unit)>()

    fun closeConversation()
    {
        onConversationFinished.forEach { it.invoke() }
    }
}
